v2/tools/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -181,3 +320,97 @@ func (i *importableARMResource) nameFromID(id *arm.ResourceID) string {
 	klog.V(3).Infof("Name: %s", id.Name)
 	return id.Name
 }
+
+func (i *importableARMResource) createContainerURI(id string, subType string) string {
+	parts := strings.Split(subType, "/")
+	return fmt.Sprintf("%s/%s", id, parts[len(parts)-1])
+}
+
+func (i *importableARMResource) SetName(
+	importable genruntime.ImportableARMResource,
+	name string,
+	owner genruntime.ResourceReference,
+) {
+	// Kubernetes names are prefixed with the owner name to avoid collisions
+	n := name
+	if owner.Name != "" {
+		n = fmt.Sprintf("%s-%s", owner.Name, name)
+	}
+
+	importable.SetName(n)
+
+	// AzureName needs to be exactly as specficied in the ARM URL
+	// Need to use reflection to set it
+	specField := reflect.ValueOf(importable.GetSpec()).Elem()
```

Comment:
very minor: You could put this into `reflecthelpers` if you wanted and use it here, although if you did I'd leave a comment there akin to the one we have in `kubernetes_resource.go`:
```
	// Some types, but not all, have a corresponding:
	// 	SetAzureName(name string)
	// They do not if the name must be a fixed value (like 'default').
```

Basically saying "don't use this on types that have a fixed value for the name".

OTOH totally fine for it to stay here as something specific to `asoctl` too, given the limited usage.

Reply:
Added to my existing TODO to move these things into reflecthelpers.